Automattic/wp-calypso
# Add date-sensitive 2021 sale banner to jetpack.com/pricing

#### Changes proposed in this Pull Request

Fixes `pbNhbs-mt-p2`.

* Add a new `NewYear2021SaleBanner` component to the pricing page header on jetpack.com.
* Conditionally render the sale banner only if the `newpack` query parameter is set or if the browser time is currently between **Jan 1 00:00:00 UTC** and **Jan 18 23:59:59 UTC**.

#### Testing instructions

Same instructions as with the previous PR #47531 (copied and updated here for easy access):

1. Checkout this pull request and start Calypso Green.
2. In your Calypso Green test environment, visit `<host>/pricing`.
3. Verify that no New Year 2021 sale banner is displayed on the page.
4. Add `?newpack` to the query string your browser's address bar and reload the page (ex: `http://jetpack.cloud.localhost:3000/pricing?newpack`).
5. Verify that the New Year 2021 sale banner is now visible.
6. Remove `?newpack` from your browser's address bar and reload the page. The banner should disappear.
7. Change your computer's system time to any date and time between **2021-01-01 00:00:00 UTC** and **2021-01-18 23:59:59 UTC** (on Mac: Apple menu > System Preferences > Date & Time).
8. Reload the page. Verify that the sale banner re-appears.
9. Verify that the banner resizes to fit both mobile and desktop layouts. The mobile banner is more vertically laid out, whereas the desktop banner is wider with a horizontal layout.
10. Verify that the following alt text is present on the banner image: "New Year 2021 sale! Save 40% at checkout with code NEWPACK through January 18"

#### Reference screenshots

##### Banner not visible

<img height="220" alt="Screen Shot 2020-12-22 at 14 08 32" src="https://user-images.githubusercontent.com/670067/102928898-31062400-445f-11eb-9657-a41fab751fc1.png"> <img height="220" alt="Screen Shot 2020-12-22 at 14 07 22" src="https://user-images.githubusercontent.com/670067/102928845-19c73680-445f-11eb-9e22-913b5f861230.png">

##### Banner visible

<img height="300" alt="Screen Shot 2020-12-22 at 13 49 24" src="https://user-images.githubusercontent.com/670067/102928101-c43e5a00-445d-11eb-907b-ee14cf2ea77a.png"> <img height="300" alt="Screen Shot 2020-12-22 at 13 48 56" src="https://user-images.githubusercontent.com/670067/102928094-c1dc0000-445d-11eb-92b6-2180d5f107fc.png">
